docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -13,11 +13,20 @@ On this page, you can learn about the following topics:
 
 ## Incremental compilation
 
-The Kotlin Gradle plugin supports incremental compilation. Incremental compilation tracks changes to files in the classpath
-between builds so that only the files affected by these changes are compiled. Incremental compilation works with [Gradle's
-build cache](#gradle-build-cache-support) and supports [compilation avoidance](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_compile_avoidance).
+The Kotlin Gradle plugin supports incremental compilation, which is enabled by default for Kotlin/JVM and Kotlin/JS projects.
+Incremental compilation tracks changes to files in the classpath between builds so that only the files affected
+by these changes are compiled.
+This approach works with [Gradle's build cache](#gradle-build-cache-support) and supports [compilation avoidance](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_compile_avoidance).
 
-Incremental compilation is supported for Kotlin/JVM and Kotlin/JS projects, and is enabled by default.
+For Kotlin/JVM, incremental compilation relies on classpath snapshots,
+which capture the API structure of modules to determine when recompilation is necessary.
+You can configure classpath snapshots as **fine-grained** or **coarse-grained**, depending on the level of detail required:
+
+* **Fine-grained snapshots** track changes within class members, such as fields or properties.
+This setting minimizes the scope of recompilation by targeting only files affected by specific member changes.
+To maintain performance, the Kotlin Gradle plugin skips creating fine-grained snapshots for `.jar` files in the Gradle cache.
+* **Coarse-grained snapshots** track changes only at the class level, resulting in smaller snapshots but a broader recompilation scope.
+This setting is useful for classes that change infrequently, such as external libraries.
```

Comment:
Same as above, it's not a setting. But more importantly, can we somehow tie this to the previous explanation?


> **Fine**
> Detecting the member-level changes allows Kotlin compiler to find and recompile the classes that might be using these members directly.
> <...>
> **Coarse**
> Compared to the fine-grained snapshot, we would not only recompile the classes which depend on the changed members, but we would also recompile all other classes which depend on the changed class.

There might be too much "class" in this sentence, but maybe you'd see a way to rephrase it better.

Reply:
Hmm 🤔 - yes it repeats class a bit often, but I don't think there is a way to avoid it without risking it becoming ambiguous.

What is your opinion about this version:

> Compared to fine-grained snapshots, coarse-grained snapshots recompile not only the classes that depend on the modified members but also all other classes that depend on the modified class.

It's a bit more explicit but removes any potential confusion from the sentence

Reply:
It feels wrong to say that "snapshots recompile something", because they don't.

How about a more basic explanation? A rough version:

> Coarse-grained snapshots store class info without member-level details. When a class-level change is detected, Kotlin compiler knows that some members have changed, but it doesn't know which exactly. So it would recompile all uses of the changed class.